src/main/java/az/ingress/ms/user/service/concrete/UserServiceHandler.java

```diff
@@ -0,0 +1,97 @@
+import az.ingress.ms.user.exception.NotFoundException;
+import az.ingress.ms.user.exception.PasswordMismatchException;
+import az.ingress.ms.user.model.request.SignInRequest;
+import az.ingress.ms.user.model.request.SignUpRequest;
+import az.ingress.ms.user.model.request.UserProfileUpdateRequest;
+import az.ingress.ms.user.model.response.SignInResponse;
+import az.ingress.ms.user.model.response.UserResponse;
+import az.ingress.ms.user.service.abstraction.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+import static az.ingress.ms.user.exception.ExceptionMessages.PASSWORD_DOES_NOT_MATCH;
+import static az.ingress.ms.user.exception.ExceptionMessages.USER_NOT_FOUND_WITH_EMAIL;
+import static az.ingress.ms.user.exception.ExceptionMessages.USER_NOT_FOUND_WITH_ID;
+import static az.ingress.ms.user.mapper.factory.SignMapper.SIGN_MAPPER;
+import static az.ingress.ms.user.mapper.factory.UserMapper.USER_MAPPER;
+import static java.util.Optional.ofNullable;
+import static lombok.AccessLevel.PRIVATE;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+@FieldDefaults(level = PRIVATE, makeFinal = true)
+public class UserServiceHandler implements UserService {
+
+    UserRepository userRepository;
+    BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @Override
+    public void createUser(SignUpRequest signUpRequest) {
+        logger.info("ActionLog.createUser.start");
+        userRepository.save(USER_MAPPER.buildUserEntity(signUpRequest));
+        logger.info("ActionLog.createUser.end");
+    }
+
+    @Override
+    public SignInResponse login(SignInRequest signInRequest) {
+        logger.info("ActionLog.login.start");
+        var user = findUserByEmail(signInRequest.getEmail());
+        verifyPassword(signInRequest.getPassword(), user.getPassword());
+        logger.info("ActionLog.login.end");
+        return SIGN_MAPPER.buildSignInResponse(user);
+    }
+
+
+    @Override
+    public void updateProfile(Long userId, UserProfileUpdateRequest updateRequest) {
+        logger.info("ActionLog.updateProfile.start");
+        var user = findUserById(userId);
+
+        ofNullable(updateRequest.getFirstName()).ifPresent(user::setFirstName);
+        ofNullable(updateRequest.getLastName()).ifPresent(user::setLastName);
+        ofNullable(updateRequest.getBirthDate()).ifPresent(user::setBirthDate);
+        ofNullable(updateRequest.getProfilePicture()).ifPresent(user::setProfilePicture);
+
+        userRepository.save(user);
+        logger.info("ActionLog.updateProfile.end");
+    }
+
+    @Override
+    public UserResponse getUserById(Long userId) {
+        return USER_MAPPER.buildUserResponse(findUserById(userId));
+    }
+
+    @Override
+    public List<UserResponse> getUsers() {
```

Comment:
marketplace user count min ~ 10k, if you get all users from db you catch OutOfMemory and so late response time